spring-integration-cassandra/src/test/java/org/springframework/integration/cassandra/config/outbound-adapter-parser.xml

```diff
@@ -0,0 +1,73 @@
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:int="http://www.springframework.org/schema/integration"
+	xmlns:util="http://www.springframework.org/schema/util"
+	xmlns:cassandra="http://www.springframework.org/schema/data/cassandra"
+	xmlns:int-cassandra="http://www.springframework.org/schema/integration/cassandra"
+	xsi:schemaLocation="http://www.springframework.org/schema/integration http://www.springframework.org/schema/integration/spring-integration.xsd
+		http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd
+		http://www.springframework.org/schema/data/cassandra http://www.springframework.org/schema/data/cassandra/spring-cassandra.xsd
+		http://www.springframework.org/schema/integration/cassandra http://www.springframework.org/schema/integration/cassandra/spring-integration-cassandra.xsd
+		http://www.springframework.org/schema/util http://www.springframework.org/schema/util/spring-util.xsd">
+
+
+    <int:poller default="true" fixed-delay="50"/>
+
+	<bean id="cassandraSession" class="org.mockito.Mockito"
+		factory-method="mock">
+		<constructor-arg value="com.datastax.driver.core.Session" />
+	</bean>
+
+	<bean id="writeOptions" class="org.springframework.cassandra.core.WriteOptions">
+		<property name="ttl" value="60" />
+		<property name="consistencyLevel">
+			<util:constant
+				static-field="org.springframework.cassandra.core.ConsistencyLevel.ONE" />
+		</property>
+		<property name="retryPolicy">
+			<util:constant
+				static-field="org.springframework.cassandra.core.RetryPolicy.DOWNGRADING_CONSISTENCY" />
+		</property>
+	</bean>
+
+	<cassandra:mapping />
+	<cassandra:converter />
+	<cassandra:template id="cassandraTemplate" />
+	
+	<int:channel id="input">
+		<int:queue/>
+	</int:channel>
+	
+	<bean id="resultChannel" class="org.springframework.integration.channel.QueueChannel"/>
+	
+	
+
+	
+	<int-cassandra:outbound-channel-adapter id="outbound1" 
+							cassandra-template="cassandraTemplate" 
+							operation-type="INSERT"
+							write-options="writeOptions"
+							auto-startup="false"
+							/>
+							
+     <int-cassandra:outbound-channel-adapter id="outbound2" 
+							channel="input"
+							cassandra-template="cassandraTemplate" 
+							operation-type="INSERT"
+							cql-ingest="insert into book (isbn, title, author, pages, saleDate, isInStock) values (?, ?, ?, ?, ?, ?)"
+							/>
+    
+    <int-cassandra:outbound-channel-adapter id="outbound3" 
+							cassandra-template="cassandraTemplate" 
+							operation-type="STATEMENT"
+							write-options="writeOptions"
+							query="SELECT * FROM book limit :size"
+                            output-channel="resultChannel"
+                            produces-reply="true"
+							auto-startup="true">
+		    <int-cassandra:parameter-expressions name="author" expression="payload"/>
+		    <int-cassandra:parameter-expressions name="size" expression="headers.limit"/>	
```

Comment:
In addition I'd like to see `author` in the `SELECT` to be sure that we are good here.
From other side you should agree with me that `SELECT` sample is exactly for the `gateway` behavior.
Although, I'm sure that we can't distinguish the `SELECT` on the adapter level to disable such kind of behavior. 
The strange result there should be signal for end user.